codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/protocols/serialize/JsonSerializerGenerator.kt

```diff
@@ -90,7 +90,7 @@ class JsonSerializerGenerator(
     private val jsonName: (MemberShape) -> String,
     private val customizations: List<JsonSerializerCustomization> = listOf(),
 ) : StructuredDataSerializerGenerator {
-    data class Context<T : Shape>(
+    data class Context<out T : Shape>(
```

Comment:
Weird syntax, but this makes `Context` covariant in `T`: it makes `Context<T> : Context<U> where T : U`.

In particular, it lets us use `Context<CollectionShape>` in a `Context<Shape>` argument.